examples/todoapp/lib/base/models/todo_model.dart

```diff
@@ -5,109 +5,130 @@
 // license that can be found in the LICENSE file or at
 // https://opensource.org/licenses/MIT.
 
-import 'package:equatable/equatable.dart';
 import 'package:json_annotation/json_annotation.dart';
+import 'package:realm/realm.dart';
 // ignore: implementation_imports
 import 'package:rx_bloc_list/src/models/identifiable.dart';
+
 part 'todo_model.g.dart';
+part 'todo_model.realm.dart';
 
-@JsonSerializable()
-class TodoModel with EquatableMixin implements Identifiable {
-  TodoModel({
-    required this.id,
-    required this.title,
-    this.createdAt = 0,
-    this.description = '',
-    this.completed = false,
-  });
+enum TodoModelActions { add, update, delete, none }
 
+@RealmModel()
+@JsonSerializable()
+class $TodoModel implements Identifiable {
   /// The unique identifier of the todo
   /// If the todo is not persisted yet, the id is null
-  final String? id;
+  @PrimaryKey()
+  @MapTo('_id')
+  late String? id;
 
   /// The title of the todo
   ///
   /// The title is required and must not be null
-  final String title;
+  late String title;
 
   /// The description of the todo
   ///
   /// The description is optional and can be null
   /// The default value is an empty string
-  final String description;
+  late String description;
 
   /// The creation date of the todo
   ///
   /// The creation date is required and must not be null
   /// The default value is false
-  final bool completed;
+  late bool completed;
 
   /// The completion status of the todo
   ///
   /// The completion status is required and must not be null
   /// The default value is false
-  final int createdAt;
+  late int createdAt;
+
+  @JsonKey(includeFromJson: false, includeToJson: false)
+  late bool? synced;
+
+  late String? action;
```

Comment:
Please document all properties

Reply:
Considering we have a enum, why we use a string here ?

Reply:
Since the realm atlas doesn't support storing enums, I'm storing the value of the enum as a String

https://github.com/realm/realm-dart/issues/681#issuecomment-2094868239